### DISCUSSION TITLE
PDF viewer for pymupdf doc objects

### DISCUSSION THREAD

**Discussion by JustSaX** [pymupdf/PyMuPDF-Utilities#44] [Q&A]
Is there a PDF viewer that can view pymupdf doc objects within Python directly, so that saving the file and opening with a viewer is not necessary?

I want to change PDFs with pymupdf and preview the changes before saving.

With doc objects I mean the following: `doc = fitz.open("my_pdf.pdf")`

Thanks!

**Comment by JorjMcKie** [comment]
You could make a pixmap of the page after your changes display its Pillow image via `img.show()` as a very basic solution.
